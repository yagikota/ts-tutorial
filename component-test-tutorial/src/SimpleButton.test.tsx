import { render, screen } from "@testing-library/react";
import { SimpleButton } from "./SimpleButton";
import userEvent from "@testing-library/user-event";

test("Clicking the button toggles the display between ON and OFF.", async () => {
    const user = userEvent.setup()
    render(<SimpleButton />);
    const simpleButton = screen.getByRole("button");
    expect(simpleButton).toHaveTextContent("OFF");
    await user.click(simpleButton); // userEvent.click(simpleButton)が実行されるまで待機
    expect(simpleButton).toHaveTextContent("ON");
});
