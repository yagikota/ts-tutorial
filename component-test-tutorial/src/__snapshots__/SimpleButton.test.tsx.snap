// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`As soon as it is drawn, it is marked OFF. 1`] = `
<div>
  <button>
    OFF
  </button>
</div>
`;
